src/actions/users.js

```diff
@@ -610,6 +610,22 @@ export function getMe() {
     };
 }
 
+export function updateServiceTermsStatus(userId, serviceTermsId, accepted) {
+    return async (dispatch, getState) => {
+        await bindClientFunc(
+            Client4.updateServiceTermsStatus,
+            UserTypes.UPDATE_SERVICE_TERMS_STATUS_REQUEST,
+            UserTypes.UPDATE_SERVICE_TERMS_STATUS_SUCCESS,
+            UserTypes.UPDATE_SERVICE_TERMS_STATUS_FAILURE,
+            userId,
+            serviceTermsId,
+            accepted
+        )(dispatch, getState);
+        await getMe()(dispatch, getState);
```

Comment:
This should be `dispatch(getMe())` but why are we getting the user profile again agter the ToS was accepted or denied?

Reply:
We need to update the current user object in the store, so that the selector, which reads `!user.latest_service_terms_accepted` works as expected.

Reply:
Also, if we don't use `await` here, the user data is not updated before the update action in the webapp returns a response without the user object being updated. It leads to `showCustomTerms` selector to behave incorrectly. So, we need to `await` in both.